### PR TITLE
feat(analytics): Add v2 payment analytics (payment-intents analytics)

### PR DESCRIPTION
**Pull request by tsdk02**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added analytics based on `payment intent`

Currently supporting 4 metrics
- successful smart retries
- total smart retries
- smart retried amount
- payment intent count

filters
- status
- currency

group by
- status
- currency

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Utilizing payment intents aligns closely with the data needs of merchants, enhancing their ability to manage and process transactions more effectively.
## How did you test it?

Tested using local data by sending cURL requests through Postman.
Tested using Postgres data from `payment_intents` table and also using Clickhouse data from `payment_intent` table.
Screenshot 2024-06-28 at 12 33 21 PM
Screenshot 2024-06-28 at 12 35 44 PM
Screenshot 2024-06-28 at 12 36 05 PM

Screenshot 2024-06-28 at 12 33 38 PM

Screenshot 2024-06-28 at 12 37 02 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/analytics/src/payment_intents/metrics/successful_smart_retries.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        filters.set_filter_clause(&mut query_builder).switch()?;
+
+        query_builder
+            .add_filter_clause("merchant_id", merchant_id)
+            .switch()?;
+        query_builder
+            .add_custom_filter_clause("attempt_count", "1", FilterTypes::Gt)
+            .switch()?;
+        query_builder
+            .add_custom_filter_clause("status", "succeeded", FilterTypes::Equal)
```
Comment (ivor-juspay) [inline crates/analytics/src/payment_intents/metrics/successful_smart_retries.rs:76]: can you use the payment intent success type here?

File: crates/analytics/src/payment_intents/metrics/smart_retried_amount.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                alias: Some("start_bucket"),
+            })
+            .switch()?;
+        query_builder
+            .add_select_column(Aggregate::Max {
+                field: "created_at",
+                alias: Some("end_bucket"),
+            })
+            .switch()?;
+
```
Comment (ivor-juspay) [inline crates/analytics/src/payment_intents/metrics/smart_retried_amount.rs:63]: is `status = succeeded` filter required here?

File: crates/analytics/src/payment_intents/metrics/smart_retried_amount.rs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        filters.set_filter_clause(&mut query_builder).switch()?;
+
```
Comment (ivor-juspay) [inline crates/analytics/src/payment_intents/metrics/smart_retried_amount.rs:65]: and this
```suggestion
        query_builder
            .add_custom_filter_clause("attempt_count", "1", FilterTypes::Gt)
            .switch()?;
```

File: crates/analytics/src/payment_intents/metrics/smart_retried_amount.rs (hunk 2 above)
Comment (tsdk02) [reply crates/analytics/src/payment_intents/metrics/smart_retried_amount.rs:63]: Yes added it

File: crates/analytics/src/payment_intents/metrics/smart_retried_amount.rs (hunk 3 above)
Comment (tsdk02) [reply crates/analytics/src/payment_intents/metrics/smart_retried_amount.rs:65]: Added this

File: crates/analytics/src/payment_intents/metrics/successful_smart_retries.rs (hunk 1 above)
Comment (tsdk02) [reply crates/analytics/src/payment_intents/metrics/successful_smart_retries.rs:76]: Done

**Review by lsampras — APPROVED** [review]

**Review by Abhitator216 — APPROVED** [review]

**Pull request merged**